File: src/LeetCode121.java
class LeetCode121 {
    public static int maxProfit(int[] prices) {
        //int result=0;
        int min=100000;
        int max=0;
        for(int i=0; i<prices.length; i++){
            min = Math.min(min, prices[i]);
            if(prices[i] > min){
                max = Math.max(max, prices[i] - min);
            }
        }
        return max;
    }
}

File: cpp/src/leetcode121.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
  int maxProfit(vector<int> &prices) {
    int m_res = 0, m_min = prices[0];
    for (int price : prices) {
      m_res = max(m_res, price - m_min);
      m_min = min(m_min, price);
    }
    return m_res;
  }
};
